plainly-aescripts/src/validation/textValidators.ts

```diff
@@ -0,0 +1,124 @@
+import { getAllComps, getTextLayersByComp } from '../utils';
+import { ProjectIssueType, type TextLayerIssues } from './types';
+
+function checkTextLayers(): TextLayerIssues[] {
+  const comps = getAllComps(app.project);
+  const textLayers: TextLayerIssues[] = [];
+
+  for (let i = 0; i < comps.length; i++) {
+    const comp = comps[i];
+    const layers = getTextLayersByComp(comp);
+    if (layers.length === 0) {
+      continue;
+    }
+
+    for (let j = 0; j < layers.length; j++) {
+      const layer = layers[j];
+      if (layer.guideLayer) {
+        continue;
+      }
+
+      const textDocument = layer.sourceText.value;
+      // check if characterRange exists in textDocument (After Effects 24.3+)
+      if (typeof textDocument.characterRange !== 'function') {
+        continue;
+      }
+
+      let hasCharacterAllCaps = false;
```

Comment:
I think you can also do this check

```
if (textDocument.allCaps) // this checks only the first char

```

for unsupported AE version and maybe to optimize a bit?